### DISCUSSION TITLE
SNR optimised reconstruction with noise matrix

### DISCUSSION THREAD

**Discussion by Laura2305** [MagneticResonanceImaging/MRIReco.jl#135] [Q&A] [answered]
Hi, I am using MRIReco.jl to reconstruct spinal cord data acquired with a multi-echo GRE. When applying a SENSE reconstruction, it is possible to take the noise matrix into account to optimise the image SNR, as shown by Pruessmann et al., in 1999. This usually does not have any effect on the image quality for brain data, but it should improve the results for spinal cord data. I have been looking at the reconstruction functions, but it seems like the noise matrix is not expected as input. Is there an implemented way to do this? I plan on attending the next developers and users meeting in December, and I would be happy to discuss this matter on that occasion.

Pruessmann, K. P., Weiger, M., Scheidegger, M. B., & Boesiger, P. (1999). SENSE: Sensitivity encoding for fast MRI. Magnetic Resonance in Medicine, 42(5), 952–962. [https://doi.org/10.1002/(SICI)1522-2594(199911)42:5<952::AID-MRM16>3.0.CO;2-S](https://doi.org/10.1002/(SICI)1522-2594(199911)42:5%3C952::AID-MRM16%3E3.0.CO;2-S)

**Comment by aTrotier** [comment]
We currently don't support that. I think the easiest way to do so will be
to perform the prewhitening on the rawdata (gadgetron is also doing that)
rather than introducing the noise matrix during the sense reconstruction

You can take a look at
http://hansenms.github.io/sunrise/sunrise2014/Hansen_ImageReconstruction_ParallelImaging_20140428.pdf





Le mer. 23 nov. 2022 à 16:18, Laura Beghini ***@***.***> a
écrit :

> Hi, I am using MRIReco.jl to reconstruct spinal cord data acquired with a
> multi-echo GRE. When applying a SENSE reconstruction, it is possible to
> take the noise matrix into account to optimise the image SNR, as shown by
> Pruessmann et al., in 1999. This usually does not have any effect on the
> image quality for brain data, but it should improve the results for spinal
> cord data. I have been looking at the reconstruction functions, but it
> seems like the noise matrix is not expected as input. Is there an
> implemented way to do this? I plan on attending the next developers and
> users meeting in December, and I would be happy to discuss this matter on
> that occasion.
>
> Pruessmann, K. P., Weiger, M., Scheidegger, M. B., & Boesiger, P. (1999).
> SENSE: Sensitivity encoding for fast MRI. Magnetic Resonance in Medicine,
> 42(5), 952–962.
> https://doi.org/10.1002/(SICI)1522-2594(199911)42:5<952::AID-MRM16>3.0.CO;2-S
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MagneticResonanceImaging/MRIReco.jl/discussions/135>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC5P7O2QGUIWHGXM4ECIUNLWJYYVHANCNFSM6AAAAAASJCRR64>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
------------------------------------------------------------
TROTIER Aurélien
31 allée de la Biotte
33470 Gujan-Mestras
06.09.12.61.61
***@***.***

**Reply by JeffFessler** [reply]
I agree.  If you need help with prewhitening in Julia, just ask.  It's basically just `inv(sqrt(covariance))` multiplied by the coil data after appropriate reshape/permutedims, where you get the noise `covariance` from a noise-only readout.

**Comment by mrikasper** [comment]
I am also interested in this question. Maybe you could help with two clarifications:

1. Great resource @aTrotier . If you look at Michael Hansen's PDF, it looks as if both the raw data _and_ the coil sensitivity maps are altered before reconstruction (see screenshots, yellow marker) by the result of the Cholesky decomposition of the Noise Covariance matrix - is that an alternative, but equivalent formulation to what you or @JeffFessler are suggesting or do we have to alter the coil sensitivity maps in either case?
    - Is there a difference in the approach whether we do self-calibrated parallel imaging recon or use a different scan (like I do for spirals) for estimating the coil sensitivity maps?
![image](https://user-images.githubusercontent.com/13321311/203633283-2efa7fe9-753e-4e2d-ab60-572e938e68ca.png)
![image](https://user-images.githubusercontent.com/13321311/203634040-8d2b07a7-8852-4039-94b1-8b12f19391cb.png)
2. For MRIReco.jl, I think the practical question is _where_ the prewhitening of data and/or sensitivity maps should happen. As a function operating on `RawAcquisitionData` or `AcquisitionData`? Or on `[Raw]AcquisitionData` and the array representing the coil sensitivity array. Maybe @migrosser or @tknopp have a preference?

**Reply by JeffFessler** [reply]
Any matrix square root (inverse) is fine, including the Cholesky version.  I would have used Cholesky via `\` if I had a single vector `b` on the right-hand side, but here we have `Nsamples` on the RHS so we end up using `inv` anyway.  Either way is fine.  (Don't use `sqrt` in Matlab BTW; you must use `sqrtm` there.  That potential confusion is enough of a reason to favor `chol`!)
You raise a crucial point about the need to also adjust the coil maps accordingly!
Once you have the sensitivity maps, I don't think the k-space sampling that led to them matters anymore.  You still must apply `L_inv`.

**Reply by aTrotier** [reply]
@mrikasper if you extract the sensitivity maps directly from the raw data with a calibration region at the center of the kspace or using a low resolution reconstruction of non Cartesian data, you don't need to scale the sensitivity maps.

**Reply by mrikasper** [reply]
Thank you for the insightful responses @JeffFessler and @aTrotier! Our spirals are uniformly undersampled, so they rely on a second scan for the maps, so for that use case, it's probably best to follow the approach that Jeff & Michael Hansen outlined, i.e., both altering sensitivity maps and data (or the imaging operator, s.b.)

**Comment by tknopp** [comment]
> Maybe @migrosser or @tknopp have a preference?

This should not be done on the `[Raw]AcquisitionData` level but when setting up the imaging operator:

https://github.com/MagneticResonanceImaging/MRIReco.jl/blob/master/src/Reconstruction/IterativeReconstruction.jl#L154

One can either do this "lazily" like we do with the density compensation weights:
https://github.com/MagneticResonanceImaging/MRIReco.jl/blob/master/src/Reconstruction/IterativeReconstruction.jl#L189
or do it directly on the data. In the later case one should copy the smaps since `reconstruction` should not mutate its input arguments.

**Reply by mrikasper** [reply]
That's great, thank you for the pointers, maybe @Laura2305 can go from there and create and issue/PR with some simulated data? Lazily and similar to the density compensation sounds good, since it's analogous to what you have been doing, right?

**Reply by Laura2305** [reply]
Thank you all for the sources and suggestions. I have time to work ion this starting from January.

**Comment by Laura2305** [comment, marked as answer]
I have implemented the raw data pre-whitening for the multi coil reconstruction, and sent a pull request #140.
Should I implement it for the multi coil multi echo reconstruction as well?

**Reply by tknopp** [reply]
> Should I implement it for the multi coil multi echo reconstruction as well?

yes, that would be great
